SpatialGDK/Source/SpatialGDK/Private/Utils/Interest/NetCullDistanceInterest.cpp

```diff
@@ -0,0 +1,334 @@
+	const TMap<float, Worker_ComponentId>& NetCullDistancesToComponentIds = InClassInfoManager->GetNetCullDistanceToComponentIds();
+
+	FrequencyToConstraintsMap FrequencyToConstraints;
+
+	for (const auto& DistanceComponentPair : NetCullDistancesToComponentIds)
+	{
+		const float MaxCheckoutRadiusMeters = NetCullDistanceInterest::NetCullDistanceSquaredToSpatialDistance(DistanceComponentPair.Key);
+
+		QueryConstraint ComponentConstraint;
+		ComponentConstraint.ComponentConstraint = DistanceComponentPair.Value;
+
+		float FullFrequencyCheckoutRadius = MaxCheckoutRadiusMeters * SpatialGDKSettings->FullFrequencyNetCullDistanceRatio;
+
+		QueryConstraint RadiusConstraint;
+		RadiusConstraint.RelativeCylinderConstraint = RelativeCylinderConstraint{ FullFrequencyCheckoutRadius };
+
+		QueryConstraint CheckoutRadiusConstraint;
+		CheckoutRadiusConstraint.AndConstraint.Add(RadiusConstraint);
+		CheckoutRadiusConstraint.AndConstraint.Add(ComponentConstraint);
+
+		AddToFrequencyConstraintMap(FullFrequencyHz, CheckoutRadiusConstraint, FrequencyToConstraints);
+
+		// Add interest query for specified distance/frequency pairs
+		for (const auto& DistanceFrequencyPair : SpatialGDKSettings->InterestRangeFrequencyPairs)
+		{
+			float CheckoutRadius = MaxCheckoutRadiusMeters * DistanceFrequencyPair.DistanceRatio;
+
+			QueryConstraint FrequencyRadiusConstraint;
+			FrequencyRadiusConstraint.RelativeCylinderConstraint = RelativeCylinderConstraint{ CheckoutRadius };
+
+			QueryConstraint FrequencyCheckoutRadiusConstraint;
+			FrequencyCheckoutRadiusConstraint.AndConstraint.Add(FrequencyRadiusConstraint);
+			FrequencyCheckoutRadiusConstraint.AndConstraint.Add(ComponentConstraint);
+
+			AddToFrequencyConstraintMap(DistanceFrequencyPair.Frequency, CheckoutRadiusConstraint, FrequencyToConstraints);
+		}
+	}
+
+	FrequencyConstraints CheckoutConstraints;
+
+	// De dupe across frequencies.
+	for (auto& FrequencyConstraintsPair : FrequencyToConstraints)
+	{
+		TSchemaOption<float> SpatialFrequency = FrequencyConstraintsPair.Key == FullFrequencyHz ? FullFrequencyOptional : TSchemaOption<float>(FrequencyConstraintsPair.Key);
+		if (FrequencyConstraintsPair.Value.Num() == 1)
+		{
+			CheckoutConstraints.Add({ SpatialFrequency, FrequencyConstraintsPair.Value[0] });
+			continue;
+		}
+		QueryConstraint RadiusDisjunct;
+		RadiusDisjunct.OrConstraint.Append(FrequencyConstraintsPair.Value);
+		CheckoutConstraints.Add({ SpatialFrequency, RadiusDisjunct });
+	}
+
+	return CheckoutConstraints;
+}
+
+void NetCullDistanceInterest::AddToFrequencyConstraintMap(const float Frequency, const QueryConstraint& Constraint, FrequencyToConstraintsMap& OutFrequencyToConstraints)
+{
+	// If there is already a query defined with this frequency, group them to avoid making too many queries down the line.
+	// This avoids any extra cost due to duplicate result types across the network if they are large.
+	TArray<QueryConstraint>& ConstraintList = OutFrequencyToConstraints.FindOrAdd(Frequency);
```

Comment:
You could replace this function with a `OutFrequencyToConstraints.FindOrAdd(Frequency).Add(Constraint)` call...
Either way

Reply:
I'd rather leave it as I like that it has a more descriptive name this way, but can change if you feel strongly